bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -297,13 +296,12 @@ public void asyncOpenLedgerNoRecovery(final long lId, final OpenCallback cb, fin
     public LedgerHandle openLedgerNoRecovery(final long lId)
             throws InterruptedException, BKException {
         SyncCounter counter = new SyncCounter();
-        counter.inc();
         new LedgerOpenOp(bkc, lId, new SyncOpenCallback(), counter)
                 .initiateWithoutRecovery();
         /*
          * Wait
          */
-        counter.block(0);
+        counter.block();
```

Comment:
For BK 4.5 we might switch to JDK8. At that point we could use `CompletableFuture<Integer>` and remove the sync counter and sync callbacks. The advantage of completable future, compared to the `CountdownLatch` would be that it can also hold the operation result or exception.